src/__tests__/deployments.helper.ts

```diff
@@ -96,9 +101,20 @@ export function getVersionsByChainIdByDeploymentMap(): VersionsByChainIdByDeploy
       }
 
       // Add the version to the map
-      for (const chainId of Object.keys(deployment.networkAddresses)) {
-        versionsByDeploymentByChainId[name][chainId] ??= [];
-        versionsByDeploymentByChainId[name][chainId]?.push(deployment.version);
+      if (typeof deployment.networkAddresses === 'object') {
+        const networkAddresses: Record<string, string> =
+          deployment.networkAddresses;
+        for (const chainId of Object.keys(networkAddresses)) {
+          if (typeof deployment.version === 'string') {
+            const version: string = deployment.version;
+            versionsByDeploymentByChainId[name][chainId] ??= [];
+            versionsByDeploymentByChainId[name][chainId]?.push(version);
+          } else {
+            throw new Error('version must be a string');
+          }
+        }
+      } else {
+        throw new Error('networkAddresses must be an object');
```

Comment:
I'd suggest using early returns and remove the unnecessary assignments.

Reply:
I've changed this to use type casting, I know this is not the ideal solution but as we manage the asset JSON files I think it could be acceptable and it simplifies the code, wdyt? A more robust solution would be building a `Deployment` entity/validator, so types are enforced during the validation process 🤔

Reply:
[5118bed](https://github.com/safe-global/safe-client-gateway/pull/1533/commits/5118beddfa2bdb98fd00114e21ae415ecfbc903e)